### PR TITLE
Use for CI reusable workflows for DRY (#350)

### PR DESCRIPTION
**Pull request by spirosoik**
#### Summary
We can use across all the repositories of plugins and apps the same reusable workflow which give us the ability to change only one file instead of multiple and apart from this the ability to use different versions as we prefer.

#### Ticket Link
Ticket: https://mattermost.atlassian.net/browse/CLD-4681

### CODE REVIEW

**Comment by toninis** [comment]
@spirosoik Are we going to inform users for rebase or Circle CI will just phase out when all PRs are merged ? It would be good to globally test this across all PRs .

**Comment by spirosoik** [comment]
@toninis we are going to update the PRs as we use / commands in github comments .eg /update-branch

**Review by toninis — APPROVED** [review]

File: .github/workflows/cd.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+name: cd
+on:
+  workflow_run:
+    workflows: ["ci"]
+    types:
+      - completed
+  push:
+    tags:
+      - "v[0-9]+.[0-9]+.[0-9]+"
```
Comment (hanzei) [inline .github/workflows/cd.yml:9]: What about RC releases like https://github.com/mattermost/mattermost-plugin-github/releases/tag/v2.1.3-rc2? Shouldn't CI/CD also trigger for them?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+name: ci
+on:
+  push:
+    branches:
+      - master
+    tags:
+      - "v[0-9]+.[0-9]+.[0-9]+"
+  pull_request:
```
Comment (hanzei) [inline .github/workflows/ci.yml:8]: Why is CI limited to only PRs? Given that it takes some time to run, it's great if CI is already "don" when a PR gets submitted so any fixes can be addressed quickly.

Also: Pushes to plugin repos happen rarely. I don't think there is a cost concern.

Comment (spirosoik) [reply .github/workflows/ci.yml:8]: Do you think should run on branch? If we send multiple commits during the day in practice we consume lots of miniutes.

File: .github/workflows/cd.yml (hunk 1 above)
Comment (spirosoik) [reply .github/workflows/cd.yml:9]: This is a great point we can definitely evolve this.

Comment (hanzei) [reply .github/workflows/cd.yml:9]: Is there a ticket for fix this issue?

File: .github/workflows/ci.yml (hunk 2 above)
Comment (hanzei) [reply .github/workflows/ci.yml:8]: I would argue that it's worth the money. Once a PR is submitted, all new commits will trigger CI anyway. This only cuts the costs for branches that are still in development. Having direct feedback from CI is helpful.

**Pull request merged**